include/Models/Languages.php

```diff
@@ -0,0 +1,1107 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\Models;
+
+use PLL_Cache;
+use PLL_Language;
+use PLL_Language_Factory;
+use PLL_Translatable_Objects;
+use WP_Error;
+use WP_Term;
+use WP_Syntex\Polylang\Options\Options;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Model for the languages.
+ *
+ * @since 3.7
+ */
+class Languages {
+	public const TRANSIENT_NAME = 'pll_languages_list';
+	private const CACHE_KEY     = 'languages';
+
+	/**
+	 * Polylang's options.
+	 *
+	 * @var Options
+	 */
+	private $options;
+
+	/**
+	 * Translatable objects registry.
+	 *
+	 * @var PLL_Translatable_Objects
+	 */
+	private $translatable_objects;
+
+	/**
+	 * Internal non persistent cache object.
+	 *
+	 * @var PLL_Cache<mixed>
+	 */
+	private $cache;
+
+	/**
+	 * Flag set to true during the language objects creation.
+	 *
+	 * @var bool
+	 */
+	private $is_creating_list = false;
+
+	/**
+	 * Tells if {@see WP_Syntex\Polylang\Models\Languages::get_languages_list()} can be used.
+	 *
+	 * @var bool
+	 */
+	private $languages_ready = false;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param Options                  $options              Polylang's options.
+	 * @param PLL_Translatable_Objects $translatable_objects Translatable objects registry.
+	 * @param PLL_Cache                $cache                Internal non persistent cache object.
+	 *
+	 * @phpstan-param PLL_Cache<mixed> $cache
+	 */
+	public function __construct( Options $options, PLL_Translatable_Objects $translatable_objects, PLL_Cache $cache ) {
+		$this->options              = $options;
+		$this->translatable_objects = $translatable_objects;
+		$this->cache                = $cache;
+	}
+
+	/**
+	 * Returns the language by its term_id, tl_term_id, slug or locale.
+	 *
+	 * @since 0.1
+	 * @since 3.4 Allow to get a language by `term_taxonomy_id`.
+	 * @since 3.7 Moved from `PLL_Model::get_language()` to `WP_Syntex\Polylang\Models\Languages::get()`.
+	 *
+	 * @param mixed $value `term_id`, `term_taxonomy_id`, `slug`, `locale`, or `w3c` of the queried language.
+	 *                     `term_id` and `term_taxonomy_id` can be fetched for any language taxonomy.
+	 *                     /!\ For the `term_taxonomy_id`, prefix the ID by `tt:` (ex: `"tt:{$tt_id}"`),
+	 *                     this is to prevent confusion between `term_id` and `term_taxonomy_id`.
+	 * @return PLL_Language|false Language object, false if no language found.
+	 */
+	public function get( $value ) {
+		if ( is_object( $value ) ) {
+			return $value instanceof PLL_Language ? $value : $this->get( $value->term_id ); // Will force cast to PLL_Language.
+		}
```

Comment:
Hopefully this should fix the PHPStan ignored error that you modified.
```suggestion
		if ( $value instanceof PLL_Language ) {
			return $value;
		}
		
		if ( $value instanceof WP_Term ) {
			return $this->get( $value->term_id ); // Will force cast to PLL_Language.
		}
```

Reply:
Done in [9a346d1](https://github.com/polylang/polylang/pull/1503/commits/9a346d1fdb65dcda6297c0852d15e292a3d41860).